gremlin/ui/device.py

```diff
@@ -30,13 +31,108 @@
 from gremlin import common, event_handler, joystick_handling, shared_state
 from gremlin.error import GremlinError
 from gremlin.intermediate_output import IntermediateOutput
-from gremlin.types import InputType
+from gremlin.types import InputType, PropertyType
+from gremlin.util import file_exists_and_is_accessible
+from gremlin.common import SingletonDecorator
+from gremlin.config import Configuration
 
 
 QML_IMPORT_NAME = "Gremlin.Device"
 QML_IMPORT_MAJOR_VERSION = 1
 
 
+@SingletonDecorator
+class DeviceDatabase:
+    """Provides button/axis to name mapping for known devices"""
+
+    def __init__(self) -> None:
+        self._load()
+
+    def _load(self) -> None:
+
+        db_file = "device_db.json"
```

Comment:
Probably will need to use `util.resource_path()` here such that the file can be found when the whole thing is "compiled" with PyInstaller.

Reply:
n/p